### PR TITLE
Net2: new network interface and process

### PR DESCRIPTION
**Pull request by BooniesFX**
1.  new p2p interface
2.  message router register  
3.  logic layer more clear

### CODE REVIEW

File: p2pserver/peer/config.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿
```
Comment (lightshine001) [inline p2pserver/peer/config.json]: Duplicated. Please remove it.

Comment (BooniesFX) [reply p2pserver/peer/config.json]: After default config fix, the file is useless . I will remove it

File: p2pserver/message/msg_pack/msg_pack.go [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+func NewBlock(bk *ct.Block) ([]byte, error) {
+	log.Debug()
+	var blk mt.Block
+	blk.Blk = *bk
+
+	tmpBuffer := bytes.NewBuffer([]byte{})
+	bk.Serialize(tmpBuffer)
+
+	p := new(bytes.Buffer)
+	err := binary.Write(p, binary.LittleEndian, tmpBuffer.Bytes())
```
Comment (laizy) [inline p2pserver/message/msg_pack/msg_pack.go]: serialize a []byte is meaningless. use `tmpBuffer` directly

File: p2pserver/message/msg_pack/msg_pack.go [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+	if err != nil {
+		log.Error("Binary Write failed at new Msg")
+		return nil, err
+	}
+
+	checkSumBuf := mt.CheckSum(p.Bytes())
+	blk.Init("block", checkSumBuf, uint32(len(p.Bytes())))
+	log.Debug("The message payload length is ", blk.Length)
+
+	m, err := blk.Serialization()
```
Comment (laizy) [inline p2pserver/message/msg_pack/msg_pack.go]: double serialization is wasteful

**Review by laizy — CHANGES_REQUESTED** [review]

File: p2pserver/message/msg_pack/msg_pack.go (hunk 2 above)
Comment (BooniesFX) [reply p2pserver/message/msg_pack/msg_pack.go]: ok, i will delete it

File: p2pserver/message/msg_pack/msg_pack.go (hunk 3 above)
Comment (BooniesFX) [reply p2pserver/message/msg_pack/msg_pack.go]: refactoring msg logic to reduce useless serialization

**Pull request closed without merging**